File: Data Structures & Algorithms/is-subsequence/submission-0.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        
        int i=0 , j=0;
        while( i < s.length() && j < t.length() ) {
            if( s[i] == t[j] ) {
                i += 1;
                j += 1;
            }
            else {
                j += 1;
            }
        }

      if( i == s.length() ) {
        return true;
      }
      else {
        return false;
      }
    }
};

File: Data Structures & Algorithms/longest-common-prefix/submission-0.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        Arrays.sort(strs);

        String s1  = strs[0];
        String s2  = strs[strs.length-1];
        String ans = "";
        int i=0;
        while( i < s1.length() ) {
            if( s1.charAt(i) == s2.charAt(i) ) {
                ans += s1.charAt(i);
            }
            else break;

            i++;
        }

        return ans;
    }
}